src/lib/reader.ts

```diff
@@ -239,6 +328,17 @@ class Reader {
     this.skipVarOctetString()
   }
 
+  /**
+   * Read a variable-length unsigned integer at the cursor position and advance the cursor.
+   */
+  readVarIntNumber (): number {
+    if (this.buffer[this.cursor] <= MAX_SAFE_BYTES) {
```

Comment:
if the length prefix were multi-byte it would exceed the max_safe_bytes so we can treat the length prefix as a single byte, but that might not be immediately apparent reading this so we should have a comment to clarify that it's on purpose

Reply:
Added a comment to clarify.